File: Java/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.util.Scanner;

public class Main {

    public static int countRepaint(char[][] board, int x, int y, char startColor){
        int count = 0;
        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                char expected = ((i + j) % 2 == 0) ? startColor : (startColor == 'W' ? 'B' : 'W');
                if (board[x + i][y + j] != expected) { count++; }
            }
        }
        return count;
    }

    public static void main(String[] args) {

        Scanner s = new Scanner(System.in);

        int M = s.nextInt();
        int N = s.nextInt();
        s.nextLine();

        char[][] board = new char[M][N]; //M*N 크기의 보드
        for (int i = 0; i < M; i++) {
            String line = s.nextLine();
            for (int j = 0; j < N; j++) {
              board[i][j] = line.charAt(j);
            }
        }

        int min = Integer.MAX_VALUE;

        for (int x = 0; x <= M - 8; x++) {
            for (int y = 0; y <= N - 8; y++) {
                int repaintW = countRepaint(board, x, y, 'W');
                int repaintB = countRepaint(board, x, y, 'B');
                min = Math.min(min, Math.min(repaintW, repaintB));
            }
        }

        System.out.print(min);

        s.close();
    }
}

File: C11/백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.c
#include <stdio.h>

int min(int A, int B){
	return (A > B) ? B : A;
}

int countRepaint(char board[][50], int x, int y, char startColor){
	int count = 0;
	for(int i = 0; i < 8; i++){
		for(int j = 0; j < 8; j++){
			char expected = ((i + j) % 2 == 0) ? startColor : (startColor == 'W' ? 'B' : 'W');
			if(board[x + i][y + j] != expected){ count++; }
		}
	}
	return count;
}

int main(void){
	
	int M, N;
	scanf("%d %d", &M, &N);
	
	char board[50][50];
	for(int i = 0; i < M; i++){
		char line[51];
		scanf("%s", line);
		for(int j = 0; j < N; j++){
			board[i][j] = line[j];
		}
	}
	
	int result = 65;
	
	for(int x = 0; x <= M - 8; x++){
		for(int y = 0; y <= N - 8; y++){
			int repaintW = countRepaint(board, x, y, 'W');
			int repaintB = countRepaint(board, x, y, 'B');
			result = min(result, min(repaintW, repaintB));
		}
	}
	
	printf("%d", result);
	
	return 0;
}